src/flows_dst.rs

```diff
@@ -51,12 +51,14 @@ impl<'a> RespondToRelocateRequests<'a> {
 
     fn consensused_expect_candidate(&mut self, candidate: Candidate) {
         match (
+            self.0.action.check_shortest_prefix(),
             self.0.action.get_waiting_candidate_info(candidate),
             self.0.action.count_waiting_proofing_or_hop(),
         ) {
-            (Some(info), _) => self.resend_relocate_response_rpc(info),
-            (_, 0) => self.add_node_and_send_relocate_response_rpc(candidate),
-            (_, _) => self.send_refuse_candidate_rpc(candidate),
+            (Some(_), _, _) => self.send_expect_candidate_rpc(candidate),
```

Comment:
I think we may be bordering on abuse of the `match` statement here. Maybe one `if` for the shortest prefix and one match would read better.

Reply:
I'll rework that.